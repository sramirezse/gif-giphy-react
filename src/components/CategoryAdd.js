import React from 'react'
import { useState } from 'react'
import PropTypes from 'prop-types'
// import { GifExpertApp } from '../GifExpertApp';
// import {setCategories} from GifExpertApp;

export const CategoryAdd = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputText =  (e)=>{
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length >= 2)
        setCategories(cats=>
            [...cats, inputValue]
        );
        // console.log(999);
        // setCategories( cats=> ['otra serie',...catss])

    }
    return (
        <div>
            <form onSubmit={handleSubmit}>
                <input type="text" value={inputValue} onChange={handleInputText} />
            </form>
        </div>
    )
}

CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired
}
