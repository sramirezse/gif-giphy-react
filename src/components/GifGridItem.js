import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url, giphyUrl }) => {
    return (
        <div id='divprincipal' className="col-lg-3 col-md-6 col-sm-12 animate__animated animate__flip">
            <div className="card card-gif" >
                <img src={url} className="card-img-top" alt={title} />
                    <div className="card-body">
                        <h5 className="card-title">{title}</h5>
                        {/* <p className="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p> */}
                        <a href={giphyUrl} className="btn btn-primary">Ver en GYPHY</a>
                    </div>
            </div>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    giphyUrl: PropTypes.string.isRequired
};
