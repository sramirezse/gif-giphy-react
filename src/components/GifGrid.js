import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/GetGifs';
import { GifGridItem } from './GifGridItem';
import propTypes from 'prop-types';

export const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([]);
    const { data:images, loading } = useFetchGifs(category);
    // console.log(loading);
    // console.log(data);
    

    return (
        <div>
            <h3 className='animate__animated animate__lightSpeedInRight'>{category}</h3>
            {loading ? <div id="loadingDiv" className="loader">Loading...</div> : ''}
            <div className="row">
            {
                // console.log('asdasdasd',data),
                images.map(img => (
                    <GifGridItem key={img.id}  {...img} />

                    
                ))
            }
            </div>



        </div>
    )
}

GifGrid.prototypes = {
    category: propTypes.string.isRequired
}

