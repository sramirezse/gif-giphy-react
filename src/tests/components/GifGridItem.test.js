import React from 'react';
import {shallow} from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem></GifGridItem>', ()=>{
    const title = 'Hola soy gok√∫';
    const url = 'https://localhost';
    const giphyUrl = 'URL';
    const className  = 'animate_flip';
    const wrapper = shallow(<GifGridItem title={title} url={url} giphyUrl={giphyUrl} />);
    
    test('Debe de mostrar el componente correctamente', ()=>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo en title', ()=>{
        const h5 = wrapper.find('h5');
        expect( h5.text() ).toBe(title);
    });

    test('Debe tener la imagen igual al url y alt de los props', ()=>{
        const img = wrapper.find('img');
        // console.log(img.props().src);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe tener animate__flip', ()=>{
        const div = wrapper.find('#divprincipal');
        const className = div.prop('className');
        expect(className.includes('animate__flip')).toBe(true)
    });

});