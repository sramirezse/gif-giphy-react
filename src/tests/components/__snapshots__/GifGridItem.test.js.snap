// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem></GifGridItem> Debe de mostrar el componente correctamente 1`] = `
<div
  className="col-lg-3 col-md-6 col-sm-12 animate__animated animate__flip"
  id="divprincipal"
>
  <div
    className="card card-gif"
  >
    <img
      alt="Hola soy gokú"
      className="card-img-top"
      src="https://localhost"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Hola soy gokú
      </h5>
      <a
        className="btn btn-primary"
        href="URL"
      >
        Ver en GYPHY
      </a>
    </div>
  </div>
</div>
`;
